index.js

```diff
@@ -101,16 +144,142 @@ async function getHubspotContacts(global, storage) {
+    }
+    const authResponse = await fetchWithRetry(requestUrl)
+    const res = await authResponse.json()
+    if (!statusOk(authResponse) || res.status === 'error') {
+        const errorMessage = res.message ?? ''
+        console.error(
+            `Unable to get deals from Hubspot. Status Code: ${authResponse.status}. Error message: ${errorMessage}`
+        )
+    }
+    if (res && res['results']) {
+        for (hubspotDeal of res['results']) {
+            exists = await storage.get(hubspotDeal['id'], false)
+            if (!exists) {
+                storage.set(hubspotDeal['id'], true)
+                console.log(`Found new deal: ${hubspotDeal['id']}`)
+            } else {
+                console.log(`Deal ${hubspotDeal['id']} already exists`)
+                continue
+            }
+            const props = hubspotDeal['properties']
+            props['name'] = props['dealname'] // set name to dealname so that it doesn't show deal id in the groups list
+            posthog.capture('$groupidentify', {
+                $group_type: config.dealsGroupType,
+                $group_key: hubspotDeal['id'],
+                $group_set: props
+            })
+            // set related company group
+            if (config.companiesGroupType && hubspotDeal['associations'] && hubspotDeal['associations']['companies']) {
+                hubspotDeal['associations']['companies']['results'].forEach((company) => {
+                    posthog.capture('$groupidentify', {
+                        $groups: {
+                            [config.companiesGroupType]: company['id'],
+                            [config.dealsGroupType]: hubspotDeal['id']
+                        }
+                    })
+                })
+            }
+        }
+    }
+    let nextDealBatch
+    res['paging'] && res['paging']['next']
+        ? (nextDealBatch = res['paging']['next']['link'] + `&${global.hubspotAuth}`)
+        : null
+    await storage.set(NEXT_DEAL_BATCH_KEY, nextDealBatch)
+}
+
+async function fetchAllCompanies(config, global, storage) {
+    if (!config.companiesGroupType) {
+        console.log('No companies group type defined. Skipping fetching companies')
+        return []
+    }
+    let requestUrl = await storage.get(NEXT_COMPANY_BATCH_KEY)
+    const properties = [
+        'name',
+        'hubspotscore',
+        'city',
+        'state',
+        'zip',
+        'country',
+        'website',
+        'industry',
+        'total_revenue',
+        'total_money_raised',
+        'hs_num_open_deals',
+        'hs_total_deal_value',
+        'num_associated_deals',
+        'annualrevenue',
+        'numberofemployees'
+    ]
+    if (!requestUrl) {
+        requestUrl = `https://api.hubapi.com/crm/v3/objects/companies?limit=100&archived=false&${
+            global.hubspotAuth
+        }&properties=${properties.join(',')}`
+    }
+    const authResponse = await fetchWithRetry(requestUrl)
```

Comment:
nit: authResponse implies to me we're just testing auth, not fetching important data